### PR TITLE
Add alias `partitionID` for function `partitionId` and add documentation

### LINKED ISSUE #1994: Document `partitionId` function.
Create documentation for the `partitionId` function. See #1833 for details on how to complete this issue.

Child issue of #1833.

### PR DESCRIPTION
**Pull request by Blargian**
Closes [#1994](https://github.com/ClickHouse/clickhouse-docs/issues/1994) as part of [1833](https://github.com/ClickHouse/clickhouse-docs/issues/1833)

### Changelog category (leave one):
- Documentation (changelog entry is not required)

### CODE REVIEW

File: docs/en/sql-reference/functions/other-functions.md [hunk 1]
```diff
@@ -2984,6 +2984,57 @@ Result:
...
+
+Query:
+
+```sql
+DROP TABLE IF EXISTS mt;
+CREATE TABLE mt (n UInt64) ENGINE=MergeTree ORDER BY n PARTITION BY n % 2;
+INSERT INTO mt SELECT * FROM numbers(10);
+SELECT partitionId(*) FROM mt;
+```
+Result:
```
Comment (alexey-milovidov) [inline docs/en/sql-reference/functions/other-functions.md:3020]: The example should be more specific to demonstrate the behavior of this function.

Comment (Blargian) [reply docs/en/sql-reference/functions/other-functions.md:3031]: @alexey-milovidov i've replaced it with a less abstract example. How does it look now?

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit b9c662e1da504c4964b67d6408ad06fe9bcc0f95 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/66163/b9c662e1da504c4964b67d6408ad06fe9bcc0f95/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/66163/b9c662e1da504c4964b67d6408ad06fe9bcc0f95/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: docs/en/sql-reference/functions/other-functions.md [hunk 2]
```diff
@@ -2984,6 +2984,66 @@ Result:
 └─────────┘
 ```
 
+## partitionId
```
Comment (rschu1ze) [inline docs/en/sql-reference/functions/other-functions.md:2987]: I played around with this function (didn't know it existed) but I could not find a real use case for it.

E.g. in https://fiddle.clickhouse.com/c70544e4-ca6a-4d8f-8162-42ef3d201da2 one could get the same result by SELECTing `_partition_id` or the partition expression directly. From a user perspective, `_partition_id` is certainly easier to use.

My proposal is to remove this function (it has never been documented) and adjust integration/test_lost_part/test.py and queries/0_stateless/01748_partition_id_pruning.sql (the only two tests using `partitionId`) to use `_partition_id`.

File: tests/queries/0_stateless/01748_partition_id_pruning.sql [hunk 3]
```diff
@@ -8,17 +8,17 @@ insert into x values (1, 1), (1, 2), (1, 3), (2, 4), (2, 5), (2, 6);
 
 set max_rows_to_read = 3;
 
-select * from x where _partition_id = partitionId(1);
+select * from x where _partition_id = '1';
 
 set max_rows_to_read = 5; -- one row for subquery + subquery
 
-select * from x where _partition_id in (select partitionId(number + 1) from numbers(1));
+select * from x where _partition_id in (select number + 1 from numbers(1));
```
Comment (tavplubix) [inline tests/queries/0_stateless/01748_partition_id_pruning.sql:15]: How do I write a similar query if the partition key is a string?

File: docs/en/sql-reference/functions/other-functions.md (hunk 2 above)
Comment (UnamedRus) [reply docs/en/sql-reference/functions/other-functions.md]: > but I could not find a real use case for it.

It's exist.
Algorithm for generation of partitionId is "kind of obscured" in CHI, so if you want to understand, in which partitionId this row will land, you forced to "research and reimplement such algo" by yourself

And this function is helpful, when you build service, which need to manipulate with partitions. for example, we done that for clickhouse-backup.
Instead you can just run simple "SELECT partitionId(...)" and be sure that you get correct result.

https://github.com/ClickHouse/ClickHouse/issues/46584#issuecomment-1530932400

**Comment by rschu1ze** [comment]
Alright, didn't expect that people use undocumented functions 😄
In that case, I'll revert the removal commits and document the function instead.

**Comment by alexey-milovidov** [comment]
It's strange that the function is named `partitionId`, not `partitionID`.
It means that the author didn't put enough attention to it:
https://pastila.nl/?0008415d/c20e6794d691cc22e1107c93ccb0f8e9#vkraKbM7zaYMgJgHGCwHrQ==

**Comment by alexey-milovidov** [comment]
@rschu1ze But we better keep it, see the result above.

**Comment by Blargian** [comment]
> It's strange that the function is named `partitionId`, not `partitionID`.
> 
> It means that the author didn't put enough attention to it:
> 
> https://pastila.nl/?0008415d/c20e6794d691cc22e1107c93ccb0f8e9#vkraKbM7zaYMgJgHGCwHrQ==

Can we add an alias `partitionID` and leave `partitionId` from the docs?

**Comment by rschu1ze** [comment]
Yes, we could do that. But in that case, let's document the `ID` version and change existing tests to `ID` too.

**Comment by rschu1ze** [comment]
Tests in previous run were green, I only made the alias case-sensitive --> merging to save some CI resources.

**Pull request merged**